rate/rate.go

```diff
@@ -0,0 +1,185 @@
+// Copyright © 2017 The Things Network
+// Use of this source code is governed by the MIT license that can be found in the LICENSE file.
+
+// Package rate implements rate counting and rate limiting.
+package rate
+
+import (
+	"strconv"
+	"sync"
+	"time"
+
+	redis "gopkg.in/redis.v5"
+)
+
+// Counter interface used in rate limiter
```

Comment:
Add a little bit more documentation on how this (non-standard) component works, e.g. what is the role of the arguments? How to pick them, etc.